torch/jit/_script.py

```diff
@@ -57,7 +57,7 @@ def _is_new_style_class(cls):
 
 
 def _compile_and_register_class(obj, rcb, qualified_name):
-    ast = get_jit_class_def(obj, obj.__name__)
+    ast = get_jit_class_def(obj, obj.__name__, parse_defaults=True)
```

Comment:
Good question, but parsing defaults all the time breaks a lot of things because this same code path is also used to compile module interfaces. Parsing defaults in that breaks a bunch of module interface tests and I considered it out of scope for this PR.

Reply:
Why do the tests fail?

Reply:
I've noticed two categories of failures:
1) The function or interface has a default argument that gets inferred to be a different type when parsed as an AST constant. For example, the function definition has argument like `lambda=0` where `lambda` is not annotated or if it is, it's below the `def` as a mypy style comment. So, JIT thinks it's a int when it is actually a float. These cases can be fixed by replacing the literals with `0.0` or `0e0`, etc.
2) The function or interface has a mutable default value for an argument, like a list or a dict. This can be fixed by assigning those to a variable and then using that variable instead, or by using `None` and then assigning the actual default value in the body. One of the issues with the former solution (assigning the defaults to a variable in the containing scope and then using that variable instead) is that they are resolved using `toSugaredValue` to a generic `PythonValue` instance that you can't use as a value when processing the assignment.

Reply:
I think those two failures are not to hard to fix, and should be in scope of this PR instead of introducing a code split